automates/text_reading/src/main/scala/org/clulab/aske/automates/OdinActions.scala

```diff
@@ -695,6 +705,53 @@ class OdinActions(val taxonomy: Taxonomy, expansionHandler: Option[ExpansionHand
     mentionsDisplayOnlyArgs
   }
 
+  def filterFunction(mentions: Seq[Mention], state: State): Seq[Mention] = {
+    val toReturn = new ArrayBuffer[Mention]()
+    val (functions, other) = mentions.partition(_.label == "Function")
+    for (f <- functions) {
+      val newInputs = new ArrayBuffer[Mention]()
+      val newOutputs = new ArrayBuffer[Mention]()
+      val newTrigger = new ArrayBuffer[Mention]()
+      for (argType <- f.arguments) {
+        val sameInterval = argType._2.groupBy(_.tokenInterval) // group by token intervals
+        for (s <- sameInterval) {
+          if (argType._1 == "input" ) {
+            if (s._2.toList.length == 1) {newInputs ++= s._2} // if there's only one input, return that
```

Comment:
@alicekwak You have several calls to `s._2.toList.length` in this block and some of them could be successive given the ordering of the conditionals. Please change this code to use a single call to this function, set the value to a variable, and then use that variable in all of your conditional checks as needed.

Reply:
@pauldhein Thanks for the comment! I've addressed this issue (I believe) by setting `s._2.toList.length` as `numOfArgs` and used the variable when needed. If this is not the change you intended with the comment, please let me know. Thank you!